src/main/java/io/anserini/fusion/FuseRuns.java

```diff
@@ -0,0 +1,237 @@
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+import org.kohsuke.args4j.ParserProperties;
+import org.kohsuke.args4j.spi.StringArrayOptionHandler;
+
+class FusedRunOutputWriter implements Closeable {
+  private final PrintWriter out;
+  private final String format;
+  private final String runtag;
+
+  public FusedRunOutputWriter(String output, String format, String runtag) throws IOException {
+    this.out = new PrintWriter(Files.newBufferedWriter(Paths.get(output), StandardCharsets.UTF_8));
+    this.format = format;
+    this.runtag = runtag;
+  }
+
+  private class Document implements Comparable<Document>{
+    String docid;
+    double score;
+
+    public Document(String docid, double score)
+    {
+      this.docid = docid;
+      this.score = score;
+    }
+
+    @Override public int compareTo(Document a)
+    {
+      return Double.compare(a.score,this.score);
+    }
+
+  }
+
+  public void writeTopic(String qid, HashMap<String,Double> results) {
+    int rank = 1;
+    ArrayList<Document> documents = new ArrayList<>();
+
+    for (Map.Entry<String, Double> entry : results.entrySet()) {
+      documents.add(new Document(entry.getKey(), entry.getValue()));
+    }
+    Collections.sort(documents);   
+    for (Document r : documents) {
+      if ("msmarco".equals(format)) {
+        // MS MARCO output format:
+        out.append(String.format(Locale.US, "%s\t%s\t%d\n", qid, r.docid, rank));
+      } else {
+        // Standard TREC format:
+        // + the first column is the topic number.
+        // + the second column is currently unused and should always be "Q0".
+        // + the third column is the official document identifier of the retrieved document.
+        // + the fourth column is the rank the document is retrieved.
+        // + the fifth column shows the score (integer or floating point) that generated the ranking.
+        // + the sixth column is called the "run tag" and should be a unique identifier for your
+        out.append(String.format(Locale.US, "%s Q0 %s %d %f %s\n", qid, r.docid, rank, r.score, runtag));
+      }
+      rank++;
+    }
+  }
+
+  @Override
+  public void close() {
+    out.flush();
+    out.close();
+  }
+}
+
+// Used to hold the score and the rank of a document
+class DocScore{
+  public Double score;
+  public int initialRank; 
+
+  public DocScore(Double score, int initialRank) {
+    this.score = score;
+    this.initialRank = initialRank;
+  }
+}
+
+public class FuseRuns {
```

Comment:
In Python, it doesn't violate best practice to have multiple classes in a single file.
In Java it's different... best practice is one class per file. If you need extra classes, organize as inner classes.

So above class should be it's own file?

Reply:
And `FusedRunOutputWriter` also.